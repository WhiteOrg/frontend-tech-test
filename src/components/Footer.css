footer {
    background: linear-gradient(to bottom, #006bbb 0%,#2b40a3 100%);
    text-align: center;
    padding: 50px 0;
}

footer h3 {
    color: #FFC135;
    font-size: 24px;
    margin-bottom: 25px;
}

.icons {
    padding: 0;
    margin: 0;
}

.icons li {
    display: inline-block;
    margin: 0 5px;
}

footer svg {
    color: #fff;
    font-size: 75px;
}
