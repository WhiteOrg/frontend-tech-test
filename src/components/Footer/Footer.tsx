import React, { FC } from "react";
import { Flex } from "reflexbox";

import { Text } from "../Text";
import { StyledBackground as Background } from "./styles";
import { ReactComponent as Twitter } from "../../images/twitter.svg";
import { ReactComponent as Facebook } from "../../images/facebook.svg";

const Footer: FC = () => {
  return (
    <Background>
      <Text color="#f9cb11" fontSize={16} fontWeight={500} mb={15}>
        Share this Tournament
      </Text>
      <Flex>
        <Flex mr="5px">
          <Facebook />
        </Flex>
        <Flex ml="5px">
          <Twitter />
        </Flex>
      </Flex>
    </Background>
  );
};

export { Footer };
