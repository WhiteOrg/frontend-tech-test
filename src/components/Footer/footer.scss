.footer {
  display: flex;
  align-items: center;
  flex-direction: column;
  height: 120px;
  background-image: linear-gradient(to top, #158dd4, #56169b);
  padding-top: 30px;
  p {
    margin: 0 0 15px;
    font-size: 16px;
    font-weight: 600;
    font-stretch: condensed;
    font-style: normal;
    line-height: normal;
    letter-spacing: normal;
    color: #f9cb11;
  }
  &__icons {
    display: flex;
    button {
      background: none;
      border: none;
      padding: 0;
      font: inherit;
      cursor: pointer;
    }
    img {
      margin: 0 5px;
    }
  }
}
