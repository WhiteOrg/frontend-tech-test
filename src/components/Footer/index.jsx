import React from 'react'
import TwitterIcon from '../../svgComponents/TwitterIcon'
import FacebookIcon from '../../svgComponents/FacebookIcon'
import styled from 'styled-components'
import { componentPadding } from '../../globalStyles'

const Container = styled.footer`
    width: 100%;
    ${componentPadding}
    padding-top: 2em;
    padding-bottom: 2em;
    text-align: center;
    color: yellow;
    background: #4579fc;
    background: linear-gradient(15deg,
        #5839ad 0%,
        #024f7c 45%,
        #0377ad 100%
    );
`

const FooterTypo = styled.p`
    text-shadow: 0 0 0px yellow;
    margin: 1em;
`

const IconsWrapper = styled.div`
    width: 110px;
    display: flex;
    flex-direction: row;
    margin: 0 auto;
    align-items: center;
    justify-content: space-between;
`

// JSX
export default function Footer(){
    return(
        <Container>
            <FooterTypo>Share this tournament</FooterTypo>
            <IconsWrapper>
                <TwitterIcon />
                <FacebookIcon />
            </IconsWrapper>
        </Container>
    )
}