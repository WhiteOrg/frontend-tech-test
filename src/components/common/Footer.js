import React from "react";

import { LinkButton } from "./index";

import facebookLogo from "../../images/facebook.svg";
import twitterLogo from "../../images/twitter.svg";

const Footer = () => {
  const twitterUrl = `#`;
  const facebookUrl = `#`;
  return (
    <footer className="bg-gradient-to-tr bg-no-repeat from-purple-600 via-indigo-400 via-indigo-700 to-blue-600">
      <div className="container mx-auto p-4">
        <p className="flex justify-center font-barlow text-yellow-300 text-xl m-0 p-0">
          Share This Tournament
        </p>
        <div className="flex justify-center">
          <LinkButton title="Twitter Url" href={twitterUrl}>
            <img src={twitterLogo} alt="twitter" className="h-10 w-10"></img>
          </LinkButton>
          <LinkButton title="Facebook Url" href={facebookUrl}>
            <img src={facebookLogo} alt="Facebook" className="h-10 w-10"></img>
          </LinkButton>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
