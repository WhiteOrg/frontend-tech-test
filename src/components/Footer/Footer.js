import React from "react";
import { Container, Title, IconsWrapper } from "./Footer.styles";
import facebookIcon from "../../images/facebook.svg";
import twitterIcon from "../../images/twitter.svg";
const Footer = () => {
  return (
    <Container>
      <Title>Share this tournament</Title>
      <IconsWrapper>
        <a href="#">
          <img src={facebookIcon} alt="facebook-icon" />
        </a>
        <a href="#">
          <img src={twitterIcon} alt="twitter-icon" />
        </a>
      </IconsWrapper>
    </Container>
  );
};

export default Footer;
