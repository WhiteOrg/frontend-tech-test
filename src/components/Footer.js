import React from "react";
import FacebookIcon from "../assets/images/facebook.svg";
import TwitterIcon from "../assets/images/twitter.svg";
import { FooterWrapper } from "../styles/footer";

const Footer = () => (
  <FooterWrapper>
    <p>Share this Tournament</p>
    <div>
      <img src={TwitterIcon} alt="twitter" />
      <img src={FacebookIcon} alt="facebook" />
    </div>
  </FooterWrapper>
);

export default Footer;
