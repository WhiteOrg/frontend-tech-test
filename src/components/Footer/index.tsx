import React from "react";
import "./footer.scss";
import twitter from "../../images/twitter.svg";
import facebook from "../../images/facebook.svg";

const Footer = () => {
  return (
    <div className="footer">
      <p>Share this tournament</p>
      <div className="footer__icons">
        <button>
          <img src={twitter} alt="twitter logo" />
        </button>
        <button>
          <img src={facebook} alt="facebook logo" />
        </button>
      </div>
    </div>
  );
};

export default Footer;
