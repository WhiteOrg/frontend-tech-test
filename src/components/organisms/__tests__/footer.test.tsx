import React from "react";
import renderer from "react-test-renderer";
import Footer from "../footer";

describe("ORGANISMS: Render testing", () => {
  test("Should render Footer", () => {
    const component = renderer.create(<Footer>Share this Tournament</Footer>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
