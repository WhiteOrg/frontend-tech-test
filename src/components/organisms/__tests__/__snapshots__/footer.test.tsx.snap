// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ORGANISMS: Render testing Should render Footer 1`] = `
<footer
  className="sc-AxhUy kOLnmY"
>
  <div
    className="sc-AxiKw gIfVsQ"
  >
    <p
      className="sc-AxjAm iSHejg"
      color="#f9cb11"
      size="medium"
    >
      Share this Tournament
    </p>
    <div
      className="sc-AxhCb gewXQy"
    >
      <img
        className="sc-AxirZ kOBdo"
        src="twitter.svg"
      />
      <img
        className="sc-AxirZ kOBdo"
        src="facebook.svg"
      />
    </div>
  </div>
</footer>
`;
